src/common/snippets/src/pass/extract_constants.cpp

```diff
@@ -0,0 +1,44 @@
+// Copyright (C) 2018-2023 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#include "snippets/pass/extract_constants.hpp"
+
+#include "openvino/opsets/opset1.hpp"
+#include "snippets/itt.hpp"
+
+
+bool ov::snippets::pass::ExtractConstants::run_on_subgraph(const std::shared_ptr<op::Subgraph>& subgraph) {
+    OV_ITT_SCOPED_TASK(ov::pass::itt::domains::SnippetsTransform, "Snippets::ExtractConstants");
+    auto body = subgraph->body_ptr();
+
+    ParameterVector new_parameters;
+    OutputVector new_external_inputs = subgraph->input_values();
+
+    for (auto& op : body->get_ops()) {
+        auto constant = ov::as_type_ptr<ov::op::v0::Constant>(op);
+        if (!constant || ov::shape_size(constant->get_shape()) == 1ul)
+            continue;
+
+        const auto child = constant->get_output_target_inputs(0).begin()->get_node()->shared_from_this();
+        if (ov::snippets::op::Subgraph::constant_input_should_be_inside_body(child))
+            continue;
+
+        auto parameter = std::make_shared<ov::op::v0::Parameter>(constant->get_element_type(), constant->output(0).get_partial_shape());
+        parameter->set_friendly_name(constant->get_friendly_name());
+        ov::copy_runtime_info(constant, parameter);
+        constant->output(0).replace(parameter->output(0));
+
+        new_external_inputs.push_back(constant);
+        new_parameters.push_back(parameter);
+    }
+
+    if (new_parameters.size() != 0) {
+        body->add_parameters(new_parameters);
+        body->validate_nodes_and_infer_types();
+        subgraph->set_arguments(new_external_inputs);
```

Comment:
That's correct. What I am saying is that all of this should be done in [Subgraph::validate_and_infer_types](https://github.com/openvinotoolkit/openvino/blob/f681909db0fbf35ad211e8a56f7db049f400cb05/src/common/snippets/src/op/subgraph.cpp#L212-L216). We already call body validation there, and there is functional to replace parameters, we need just to ad an `If` to call `body->add_parameters()` if new number of paramters is larger than the old one. 
So this code on L37-39 will look like:
` subgraph->set_arguments(new_external_inputs);
  subgraph->validate_and_infer_types();`
 
This way we won't have to manually handle cases when we add or replace parameters, all this logic must be inside  `subgraph->validate_and_infer_types()`

Reply:
Please let me know, if my comment is still not clear enough, we'll discuss it offline

Reply:
Discussed offline: new `Parameter` ops creation should be outside of `validate_and_infer_types` since there is no enough information which new inputs of subgraph after `subgraph->set_arguments(new_external_inputs);` call should be replaced by new `Parameter` ops.
Conclusion: leave it so
Thanks!